doc/collision_detection.md

```diff
@@ -0,0 +1,212 @@
+
+The collision detection system supports three different types of shapes that you can build hitboxes
+from, these shapes are Polygon, Rectangle and Circle. A hitbox can be represented by many shapes to
+form the area which can be used to either detect collisions or whether it contains a point or not,
+the latter is very useful for accurate gesture detection. The collision detection does not handle
+what should happen when two hitboxes collide, so it is up to the user to implement what will happen
+when for example two position components have intersecting hitboxes.
+
+## Mixins
+### Hitbox
+The `Hitbox` mixin is mainly used for two things; to make detection of collisions with other
+hitboxes and gestures on top of your `PositionComponent`s more accurate. Say that you have a fairly
+round rock as a `SpriteComponent` for example, then you don't want to register input that is in the
+corner of the image where the rock is not displayed, since an image is always rectangular. Then you
+can use the `Hitbox` mixin to define a more accurate polygon (or another shape) for which the input
+should be within for the event to be registered on your component.
+
+You can add new shapes to the `Hitbox` just like they are added in the below `Collidable` example.
+
+### Collidable
+The `Collidable` mixin is added to a `PositionComponent` that has a `HitBox` and it is used for
+detecting collisions with other `Collidable`s. To make your component collidable you would start off
+like this:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  MyCollidable (
+    // This could also be done in onLoad instead of in the constructor
+    final shape = HitboxPolygon([
+      Vector2(0, 1),
+      Vector2(1, 0),
+      Vector2(0, -1),
+      Vector2(-1, 0),
+    ]);
+    addShape(shape);
+  }
+}
+```
+
+The `HitboxPolygon` added to the `Collidable` here is a diamond shape(◇).
+More about how the different shapes are defined in the [Shapes](#Shapes) section.
+
+Remember that you can add as many `HitboxShape`s as you want to your `Collidable` to make up more
+complex hitboxes. For example a snowman with a hat could be represented by three `HitboxCircle`s and
+a `HitboxPolygon` as its hat.
+
+To react to a collision you should override the `collisionCallback` in your component.
+Example:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  ...
+
+  @override
+  void onCollision(Set<Vector2> points, Collidable other) {
+    if (other is CollidableScreen) {
+      ...
+    } else if (other is YourOtherCollidable) {
+      ...
+    }
+  }
+}
+```
+
+In this example it can be seen how the Dart `is` keyword is used to check which other `Collidable`
+that your component collided with. The set of points is where the hitboxes of the collidables
+intersected. Note that the `onCollision` method will be called on both collidable components if they
```

Comment:
> The set of points is where the hitboxes of the collidables intersected

I can only image you mean `The set of points where the edges of the hitboxes collided`? otherwise that could be infinite

Reply:
True, it depends on if you define the hitbox as being filled or not, which I didn't.
But your explanation is more clear, so I'll put that one in there.